### PR TITLE
Upgrade explorer-schema onetable/enum to Octane

### PR DESCRIPTION
**Pull request by janzenisaac**

### CODE REVIEW

File: assets/javascripts/discourse/components/explorer-schema/one-table.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+  get styles() {
+    return this.open ? "open" : "";
+  }
+
+  @bind
+  setTableOpen(value) {
+    event.preventDefault();
+    this.open = !value;
+  }
```
Comment (CvX) [inline assets/javascripts/discourse/components/explorer-schema/one-table.js:16]: If the only thing passed to this action is `this.open` I'd rename it to e.g. `toggleOpen` and remove the argument.

File: assets/javascripts/discourse/components/explorer-schema/one-table.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import Component from "@glimmer/component";
+import { bind } from "discourse-common/utils/decorators";
+import { tracked } from "@glimmer/tracking";
+
+export default class ExplorerSchemaOneTable extends Component {
```
Comment (CvX) [inline assets/javascripts/discourse/components/explorer-schema/one-table.js:5]:
```suggestion
export default class OneTable extends Component {
```

File: assets/javascripts/discourse/components/explorer-schema/enum-info.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import Component from "@glimmer/component";
+
+export default class ParamInput extends Component {
```
Comment (CvX) [inline assets/javascripts/discourse/components/explorer-schema/enum-info.js:3]:
```suggestion
export default class EnumInfo extends Component {
```

(or change the file name 😛)

File: assets/javascripts/discourse/components/explorer-schema/enum-info.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  get enuminfo() {
+    const hash = this.args.col.enum;
+    let result = [];
+    for (let key in hash) {
+      if (!hash.hasOwnProperty(key)) {
+        continue;
+      }
+      result.push({ value: key, name: hash[key] });
+    }
+    return result;
```
Comment (CvX) [inline assets/javascripts/discourse/components/explorer-schema/enum-info.js:13]:
```suggestion
    return Object
      .entries(this.args.col.enum)
      .map(([value, name]) => { value, name });
```

Comment (janzenisaac) [reply assets/javascripts/discourse/components/explorer-schema/enum-info.js]: This is causing a https://eslint.org/docs/latest/rules/no-sequences linting error. If I understand this correctly, I don't think we will be able to do the computation of `name` within the comma operator here.

File: assets/javascripts/discourse/components/explorer-schema/enum-info.js [hunk 5]
```diff
@@ -0,0 +1,9 @@
+import Component from "@glimmer/component";
+
+export default class EnumInfo extends Component {
+  get enuminfo() {
+    return Object.entries(this.args.col.enum).map(([value, name]) => {
+      value, name;
+    });
```
Comment (CvX) [inline assets/javascripts/discourse/components/explorer-schema/enum-info.js:7]: Whoops, I was missing parens:

```suggestion
    return Object.entries(this.args.col.enum).map(([value, name]) => ({
      value, name;
    }));
```

**Review by CvX — APPROVED** [review]

**Pull request merged**